src/main.rs

```diff
@@ -167,12 +168,19 @@ fn main() -> Result<(), failure::Error> {
     } else if let Some(matches) = matches.subcommand_matches("generate") {
         let name = matches.value_of("name").unwrap_or("worker");
         let project_type = match matches.value_of("type") {
-            Some(s) => Some(settings::project::ProjectType::from_str(&s.to_lowercase())?),
+            Some(s) => Some(ProjectType::from_str(&s.to_lowercase())?),
             None => None,
         };
-        let template = matches
-            .value_of("template")
-            .unwrap_or("https://github.com/cloudflare/worker-template");
+
+        let default_template = "https://github.com/cloudflare/worker-template";
+        let template = matches.value_of("template").unwrap_or(match project_type {
+            Some(ref pt) => match pt {
+                ProjectType::Rust => "https://github.com/cloudflare/rustwasm-worker-template",
+                _ => default_template,
+            },
+            _ => default_template,
+        });
```

Comment:
cleaning up the above would allow you to define the default template as a single match against project type, and then put that in the `unwrap_or()`:

```
let default_template = match project_type {
    ProjectType::Rust => "https://github.com/cloudflare/rustwasm-worker-template",
    _ => "https://github.com/cloudflare/worker-template";
}

let template = matches.value_of("template").unwrap_or(default_template);
```